src/resources/js/commerce/gateway/stripe/checkout.js

```diff
@@ -0,0 +1,278 @@
onment. `ReadableStream` is missing.");return this._stream(t.clone(),this._options.onDownloadProgress)}return t},r=this._options.retry.methods.includes(this.request.method.toLowerCase())?this._retry(s):s();for(const[t,s]of Object.entries(responseTypes))r[t]=async()=>{this.request.headers.set("accept",this.request.headers.get("accept")||s);const o=(await r).clone();if("json"===t){if(204===o.status)return"";if(e.parseJson)return e.parseJson(await o.text())}return o[t]()};return r}_calculateRetryDelay(t){if(this._retryCount++,this._retryCount<this._options.retry.limit&&!(t instanceof TimeoutError)){if(t instanceof HTTPError){if(!this._options.retry.statusCodes.includes(t.response.status))return 0;const e=t.response.headers.get("Retry-After");if(e&&this._options.retry.afterStatusCodes.includes(t.response.status)){let t=Number(e);return Number.isNaN(t)?t=Date.parse(e)-Date.now():t*=1e3,void 0!==this._options.retry.maxRetryAfter&&t>this._options.retry.maxRetryAfter?0:t}if(413===t.response.status)return 0}return.3*2**(this._retryCount-1)*1e3}return 0}_decorateResponse(t){return this._options.parseJson&&(t.json=async()=>this._options.parseJson(await t.text())),t}async _retry(t){try{return await t()}catch(e){const s=Math.min(this._calculateRetryDelay(e),2147483647);if(0!==s&&this._retryCount>0){await delay(s);for(const t of this._options.hooks.beforeRetry){if(await t({request:this.request,options:this._options,error:e,retryCount:this._retryCount})===stop)return}return this._retry(t)}if(this._options.throwHttpErrors)throw e}}async _fetch(){for(const t of this._options.hooks.beforeRequest){const e=await t(this.request,this._options);if(e instanceof Request){this.request=e;break}if(e instanceof Response)return e}return!1===this._options.timeout?this._options.fetch(this.request.clone()):(t=this.request.clone(),e=this.abortController,s=this._options,new Promise(((r,o)=>{const i=setTimeout((()=>{e&&e.abort(),o(new TimeoutError(t))}),s.timeout);s.fetch(t).then(r).catch(o).then((()=>{clearTimeout(i)}))})));var t,e,s}_stream(t,e){const s=Number(t.headers.get("content-length"))||0;let r=0;return new globalThis.Response(new globalThis.ReadableStream({async start(o){const i=t.body.getReader();e&&e({percent:0,transferredBytes:0,totalBytes:s},new Uint8Array),await async function t(){const{done:n,value:a}=await i.read();if(n)o.close();else{if(e){r+=a.byteLength;e({percent:0===s?0:r/s,transferredBytes:r,totalBytes:s},a)}o.enqueue(a),await t()}}()}}))}}const validateAndMerge=(...t)=>{for(const e of t)if((!isObject(e)||Array.isArray(e))&&void 0!==e)throw new TypeError("The `options` argument must be an object");return deepMerge({},...t)},createInstance=t=>{const e=(e,s)=>new Ky(e,validateAndMerge(t,s));for(const s of requestMethods)e[s]=(e,r)=>new Ky(e,validateAndMerge(t,r,{method:s}));return e.HTTPError=HTTPError,e.TimeoutError=TimeoutError,e.create=t=>createInstance(validateAndMerge(t)),e.extend=e=>createInstance(validateAndMerge(t,e)),e.stop=stop,e},ky=createInstance();
```

Comment:
Do we need to include this as part of our lib? I guess we should be able to consume it from somewhere else.

Reply:
no. I added this here because @bordoni's branch was referencing a `tec-ky` library that I couldn't find. Once we fix the dependency tree in `Assets.php` we can remove this.